utils/mdldec/mdldec.c

```diff
@@ -0,0 +1,301 @@
+		}
+
+		if( counter > 1 )
+		{
+			printf( "WARNING: Sequence name \"%s\" is repeated %i times.\n", seqdesc->label, counter );
+
+			Q_snprintf( seqdesc->label, sizeof( seqdesc->label ), "%s_1", seqdesc->label );
+
+			hasduplicates = true;
+		}
+	}
+
+	if( hasduplicates )
+		puts( "WARNING: Added numeric suffix to repeated sequence name(s)." );
+}
+
+/*
+============
+BoneNameFix
+============
+*/
+static void BoneNameFix( void )
+{
+	int		 i, counter = 0;
+	mstudiobone_t	*bone;
+
+	for( i = 0; i < model_hdr->numbones; i++ )
+	{
+		bone = (mstudiobone_t *)( (byte *)model_hdr + model_hdr->boneindex ) + i;
+
+		if( bone->name[0] == '\0' )
+			Q_sprintf( bone->name, "MDLDEC_Bone%i", ++counter );
+	}
+
+	if( counter )
+		printf( "WARNING: Gived name to %i unnamed bone(s).\n", counter );
+}
+
+/*
+============
+LoadMDL
+============
+*/
+static qboolean LoadMDL( const char *modelname )
+{
+	int		 i;
+	size_t		 len;
+	char		 texturename[MAX_SYSPATH], seqgroupname[MAX_SYSPATH];
+	const char	*ext;
+	const char	 id_mdlhdr[] = {'I', 'D', 'S', 'T'};
+	const char	 id_seqhdr[] = {'I', 'D', 'S', 'Q'};
+	const char	 mdl_ver[]   = {0x0a, '\0', '\0', '\0'};
+
+	printf( "MDL: %s\n", modelname );
+
+	len = Q_strlen( modelname );
+
+	if( len > MAX_SYSPATH - 3 )
+	{
+		fputs( "ERROR: Source path is too long.\n", stderr );
+		return false;
+	}
+
+	ext = COM_FileExtension( modelname );
+
+	if( !ext )
+	{
+		fprintf( stderr, "ERROR: Source file does not have extension.\n" );
+		return false;
+	}
+
+	if( Q_strcmp( ext, "mdl" ) )
+	{
+		fprintf( stderr, "ERROR: Only .mdl-files is supported.\n" );
+		return false;
+	}
+
+	model_hdr = (studiohdr_t *)LoadFile( modelname );
+
+	if( !model_hdr )
+	{
+		fprintf( stderr, "ERROR: Can't open %s\n", modelname );
+		return false;
+	}
+
+	if( memcmp( &model_hdr->ident, id_mdlhdr, sizeof( id_mdlhdr ) ) )
+	{
+		if( !memcmp( &model_hdr->ident, id_seqhdr, sizeof( id_seqhdr ) ) )
+			fprintf( stderr, "ERROR: %s is not a main HL model file.\n", modelname );
+		else
+			fprintf( stderr, "ERROR: %s is not a valid HL model file.\n", modelname );
+
+		return false;
+	}
+
+	if( memcmp( &model_hdr->version, mdl_ver, sizeof( mdl_ver ) ) )
+	{
+		fprintf( stderr, "ERROR: %s has unknown Studio MDL format version.\n", modelname );
+		return false;
+	}
+
+	if( !model_hdr->numbodyparts )
+	{
+		fprintf( stderr, "ERROR: %s is not a main HL model file.\n", modelname );
+		return false;
+	}
+                        
+	if( destdir[0] != '\0' )
+	{
+		if( !IsFileExists( destdir ) )
+		{
+			fprintf( stderr, "ERROR: Couldn't find directory %s\n", destdir );
+			return false;
+		}
+
+		PathSlashFix( destdir );
+	}
+	else
+		COM_ExtractFilePath( modelname, destdir );
+
+	len -= 4; // path length without extension
+
+	if( !model_hdr->numtextures )
+	{
+		Q_strcpy( texturename, modelname );
+		Q_strcpy( &texturename[len], "t.mdl" );
+
+		texture_hdr = (studiohdr_t *)LoadFile( texturename );
+#ifndef  _WIN32
+		if( !texture_hdr )
+		{
```

Comment:
You can hide only one line if( !texture_hdr ), as just code blocks are allowed, i.e.
```
#if SMTH
if( smth )
#endif
{
...
}
```

Reply:
Also if we are using libpublic anyway, why not to check for XASH_WIN32?